### ISSUE TITLE
Permission Denied (publickey)

### ISSUE THREAD

**Issue by Zleander** [JonathanWenger/itergp#2]
When trying to clone ```probnum``` within the pip install, I'm running into _Permission Denied (publickey)_   

I just copied the entire command line output here, I hope it's somewhat readable
```(Numerics) ➜  Python``` is the name of the conda env and directory - commands follow after it
```zsh
(Numerics) ➜  Python pip uninstall probnum
WARNING: Skipping probnum as it is not installed.
(Numerics) ➜  Python git clone https://github.com/JonathanWenger/itergp.git
Cloning into 'itergp'...
remote: Enumerating objects: 158, done.
remote: Counting objects: 100% (17/17), done.
remote: Compressing objects: 100% (11/11), done.
remote: Total 158 (delta 4), reused 10 (delta 4), pack-reused 141
Receiving objects: 100% (158/158), 5.73 MiB | 20.79 MiB/s, done.
Resolving deltas: 100% (20/20), done.
(Numerics) ➜  Python cd itergp 
(Numerics) ➜  itergp git:(main) pip install .
Processing /Users/leander/Documents/Coding/Libraries/Python/itergp
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting probnum@ git+ssh://git@github.com/marvinpfoertner/probnum.git@ad-prototype
  Cloning ssh://****@github.com/marvinpfoertner/probnum.git (to revision ad-prototype) to /private/var/folders/v_/ht9_s7xd68bfqtfbsh60mrd00000gn/T/pip-install-wjxbkv98/probnum_61aaf70cb5ec4e819182d3d817d3cfd1
  Running command git clone --filter=blob:none --quiet 'ssh://****@github.com/marvinpfoertner/probnum.git' /private/var/folders/v_/ht9_s7xd68bfqtfbsh60mrd00000gn/T/pip-install-wjxbkv98/probnum_61aaf70cb5ec4e819182d3d817d3cfd1
  git@github.com: Permission denied (publickey).
  fatal: Could not read from remote repository.

  Please make sure you have the correct access rights
  and the repository exists.
  error: subprocess-exited-with-error
  
  × git clone --filter=blob:none --quiet 'ssh://****@github.com/marvinpfoertner/probnum.git' /private/var/folders/v_/ht9_s7xd68bfqtfbsh60mrd00000gn/T/pip-install-wjxbkv98/probnum_61aaf70cb5ec4e819182d3d817d3cfd1 did not run successfully.
  │ exit code: 128
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× git clone --filter=blob:none --quiet 'ssh://****@github.com/marvinpfoertner/probnum.git' /private/var/folders/v_/ht9_s7xd68bfqtfbsh60mrd00000gn/T/pip-install-wjxbkv98/probnum_61aaf70cb5ec4e819182d3d817d3cfd1 did not run successfully.
│ exit code: 128
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by JonathanWenger** [comment]
Thanks for the detailed bug report!

# TL;DR
Try to clone the repository again. The bug hopefully should be fixed now. If not, feel free to reopen.

# Detailed Explanation
This happened because `itergp` depends on a not-yet-released version of `probnum` that is only available as a GitHub repository. Internally the package tries to do
```
pip install git+ssh://git@github.com/marvinpfoertner/probnum.git@ad-prototype
```
So this command above should produce a similar error message. Github doesn't allow installs via pip in this manner without having deposited a public key in your Github profile. I replaced the installation in the dependencies with
```
pip install git+https://github.com/marvinpfoertner/probnum.git@ad-prototype
```
which should be possible for anybody.

**Comment by yksu** [comment]
I download the new package for itergp, after execute the command pip install. 
It shows the error "error: subprocess-exited-with-error".

The complete execution show below: 

Processing /Users/suyunkuan/Desktop/ML_Numerics for Machine Learning/Exercise/Ex2/itergp
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting probnum@ git+https://github.com/marvinpfoertner/probnum.git@ad-prototype
  Cloning https://github.com/marvinpfoertner/probnum.git (to revision ad-prototype) to /private/var/folders/vw/0wr8h_mx7t754p2l19b2mc_r0000gn/T/pip-install-ibzys1gc/probnum_ba67848f910146c89ae6e3683faa4cbe
  Running command git clone --filter=blob:none --quiet https://github.com/marvinpfoertner/probnum.git /private/var/folders/vw/0wr8h_mx7t754p2l19b2mc_r0000gn/T/pip-install-ibzys1gc/probnum_ba67848f910146c89ae6e3683faa4cbe
  Running command git checkout -b ad-prototype --track origin/ad-prototype
  Switched to a new branch 'ad-prototype'
  branch 'ad-prototype' set up to track 'origin/ad-prototype'.
  Resolved https://github.com/marvinpfoertner/probnum.git to commit bff243f58f2a7f5aa9060a973559b3b4e6ef6db1
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [47 lines of output]
      Error in sitecustomize; set PYTHONVERBOSE for traceback:
      AssertionError:
      /opt/homebrew/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py:108: _BetaConfiguration: Support for `[tool.setuptools]` in `pyproject.toml` is still *beta*.
        warnings.warn(msg, _BetaConfiguration)
      Traceback (most recent call last):
        File "/opt/homebrew/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
          main()
        File "/opt/homebrew/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/opt/homebrew/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/build_meta.py", line 338, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/build_meta.py", line 320, in _get_build_requires
          self.run_setup()
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/build_meta.py", line 335, in run_setup
          exec(code, locals())
        File "<string>", line 33, in <module>
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/__init__.py", line 87, in setup
          return distutils.core.setup(**attrs)
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 159, in setup
          dist.parse_config_files()
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/dist.py", line 868, in parse_config_files
          pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 62, in apply_configuration
          config = read_configuration(filepath, True, ignore_option_errors, dist)
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 140, in read_configuration
          return expand_configuration(asdict, root_dir, ignore_option_errors, dist)
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 195, in expand_configuration
          return _ConfigExpander(config, root_dir, ignore_option_errors, dist).expand()
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 242, in expand
          self._expand_all_dynamic(dist, package_dir)
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 286, in _expand_all_dynamic
          version=self._obtain_version(dist, package_dir),
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 332, in _obtain_version
          return _expand.version(self._obtain(dist, "version", package_dir))
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 321, in _obtain
          return self._expand_directive(
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 315, in _expand_directive
          return _expand.read_attr(directive["attr"], package_dir, root_dir)
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/config/expand.py", line 187, in read_attr
          spec = _find_spec(module_name, path)
        File "/opt/homebrew/lib/python3.10/site-packages/setuptools/config/expand.py", line 199, in _find_spec
          spec = spec or importlib.util.find_spec(module_name)
        File "/opt/homebrew/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/importlib/util.py", line 94, in find_spec
          parent = __import__(parent_name, fromlist=['__path__'])
      ModuleNotFoundError: No module named 'probnum'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
